bin/check-jenkins-health.rb

```diff
@@ -75,19 +75,41 @@ class JenkinsMetricsHealthChecker < Sensu::Plugin::Check::CLI
          proc: proc(&:to_i),
          default: 5
 
+  option :verbose,
+         short: '-v',
+         long: '--verbose',
+         boolean: true,
+         description: 'Return more verbose errors',
+         default: false
+
   def run
     https ||= config[:https] ? 'https' : 'http'
     testurl = "#{https}://#{config[:server]}:#{config[:port]}#{config[:uri]}"
 
-    r = if config[:https] && config[:insecure]
-          RestClient::Resource.new(testurl, timeout: config[:timeout], verify_ssl: false).get
-        elsif config[:https]
-          RestClient::Resource.new(testurl, timeout: config[:timeout], verify_ssl: true).get
-        else
-          RestClient::Resource.new(testurl, timeout: config[:timeout]).get
-        end
+    begin
+      r = if config[:https] && config[:insecure]
+            RestClient::Resource.new(testurl, timeout: config[:timeout], verify_ssl: false)
+          elsif config[:https]
+            RestClient::Resource.new(testurl, timeout: config[:timeout], verify_ssl: true)
+          else
+            RestClient::Resource.new(testurl, timeout: config[:timeout]).get
+          end
+      if config[:verbose]
+        r.get
+      else
+        r.get { |response| response }
+      end
+    rescue RestClient::SSLCertificateNotVerified => e
+      critical "ssl verification failed: #{e.response}"
+    rescue RestClient::ServerBrokeConnection
+      critical "server broke the connection: #{e}"
+    rescue RestClient::RequestFailed => e
+      critical "request failed: #{e.response}"
+    rescue RestClient::ExceptionWithResponse => e
+      critical "failed to #{e.response}"
+    end
 
-    if [200, 500].include? r.code
+    if r.code == 200
```

Comment:
revert back
```
[200, 500].include?
```

 r.get { |response| response }
this code doesn't raise errror, but code will be 500

Reply:
ah makes sense

Reply:
I added a comment to explain when that code applies and when it does not.